src/math/smek_mat4.cpp

```diff
@@ -0,0 +1,123 @@
+#include "smek_mat4.h"
+#include "../test.h"
+#include "smek_math.h"
+
+// Helper functions
+static bool close_enough(const Mat &a, const Mat &b, real r=0.001) {
+    bool success = true;
+    for (u32 i = 0; i < 4; i++) {
+        for (u32 j = 0; j < 4; j++) {
+            success &= Math::close_enough(a._[i][j], b._[i][j], r);
+        }
+    }
+    return success;
+}
+
+#include "../util/log.h"
+static void log_matrix(const Mat &m) {
+    LOG("\n%.2f %.2f %.2f %.2f\n%.2f %.2f %.2f %.2f\n%.2f %.2f %.2f %.2f\n%.2f %.2f %.2f %.2f",
+        m._[0][0], m._[0][1], m._[0][2], m._[0][3],
+        m._[1][0], m._[1][1], m._[1][2], m._[1][3],
+        m._[2][0], m._[2][1], m._[2][2], m._[2][3],
+        m._[3][0], m._[3][1], m._[3][2], m._[3][3]);
+}
```

Comment:
When we write a printf that knows about vectors this should be moved there as well.

(Not in this PR, just leaving this here so we don't forget.)